# Descarga de datos: pacientes con problemas de hígado

**Autor:** Mariana Bedoya Arismendy  
**Fecha:** 2026-08-16  
**Issue:** 1 - Descarga de los datos

Este notebook documenta la fuente de referencia de los datos y carga el dataset proporcionado para el ejercicio académico. El archivo utilizado en todo el proyecto corresponde a la versión entregada por el docente y se encuentra almacenado localmente en `data/01_raw/Pacientes_porblemas_higado_india.csv`.

## 📚 Importar Librerías

In [2]:
from pathlib import Path

import pandas as pd

## Fuente de los datos

- **Fuente de referencia:** UCI Machine Learning Repository, *ILPD (Indian Liver Patient Dataset)*.
- **URL de referencia:** https://archive.ics.uci.edu/static/public/225/ilpd+indian+liver+patient+dataset.zip
- **Referencia:** Dua, D. and Graff, C. (2019), UCI Machine Learning Repository.
- **Uso en este proyecto:** el docente proporcionó una versión modificada del dataset original para efectos académicos.
- **Dataset utilizado:** `data/01_raw/Pacientes_porblemas_higado_india.csv`.

El dataset original disponible en UCI **no se descarga ni se utiliza directamente**.

## 💾 Dataset utilizado

El archivo cargado corresponde exclusivamente al dataset proporcionado por el docente para el desarrollo del ejercicio académico. Este archivo se encuentra en `data/01_raw/Pacientes_porblemas_higado_india.csv` y es el dataset que se utilizará en las etapas posteriores del proyecto.

La fuente UCI se mantiene únicamente como referencia del origen del dataset base; no se realiza ninguna descarga del dataset original.

In [3]:
# Busca automáticamente la raíz del proyecto.
for candidate in [Path.cwd(), *Path.cwd().parents]:
    if (candidate / "data" / "01_raw").is_dir():
        PROJECT_ROOT = candidate
        break
else:
    raise FileNotFoundError("No se encontró la raíz del proyecto")

# Define la carpeta donde se encuentran los datos RAW.
RAW_DIR = PROJECT_ROOT / "data" / "01_raw"

# Define la ruta del dataset académico proporcionado por el docente.
LOCAL_DATA_PATH = RAW_DIR / "Pacientes_porblemas_higado_india.csv"

# Verifica que el dataset exista antes de intentar cargarlo.
if not LOCAL_DATA_PATH.exists():
    raise FileNotFoundError(f"No existe el dataset académico: {LOCAL_DATA_PATH}")

# Carga el dataset local en un DataFrame de pandas.
pacientes_df = pd.read_csv(LOCAL_DATA_PATH, low_memory=False)

# Muestra la ubicación del archivo y sus dimensiones para verificar la carga.
print(f"Archivo cargado: {LOCAL_DATA_PATH.relative_to(PROJECT_ROOT)}")
print(f"Filas: {pacientes_df.shape[0]} | Columnas: {pacientes_df.shape[1]}")

# Muestra información general sobre las columnas y sus tipos de datos.
pacientes_df.info()

Archivo cargado: data/01_raw/Pacientes_porblemas_higado_india.csv
Filas: 663 | Columnas: 1024
<class 'pandas.DataFrame'>
RangeIndex: 663 entries, 0 to 662
Columns: 1024 entries, Age to Unnamed: 1023
dtypes: float64(931), str(93)
memory usage: 5.2 MB


## 📊 Visualización de la data

In [8]:
pacientes_df.sample(10)

,Age,Gender,Total_Bilirubin,Direct_Bilirubin,Alkaline_Phosphotase,Alamine_Aminotransferase,Aspartate_Aminotransferase,Total_Protiens,Albumin,Albumin_and_Globulin_Ratio,...,Unnamed: 1014,Unnamed: 1015,Unnamed: 1016,Unnamed: 1017,Unnamed: 1018,Unnamed: 1019,Unnamed: 1020,Unnamed: 1021,Unnamed: 1022,Unnamed: 1023
456,32.0,Male,0.7,0.2,189.0,22.0,43.0,7.4,3.1,0.70,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
337,75.0,Male,2.5,1.2,375.0,85.0,68.0,6.4,2.9,0.80,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
420,55.0,Female,10.9,5.1,1350.0,48.0,57.0,6.4,2.3,0.50,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
209,45.0,Female,0.9,0.3,189.0,23.0,33.0,6.6,3.9,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
96,39.0,Male,3.8,1.5,298.0,102.0,630.0,7.1,3.3,0.80,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
615,21.0,Male,3.9,1.8,150.0,36.0,27.0,6.8,3.9,1.34,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
588,38.0,Male,1.1,0.3,198.0,86.0,150.0,6.3,3.5,1.20,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
544,54.0,Female,5.5,3.2,350.0,67.0,42.0,7.0,3.2,0.80,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
400,66.0,Female,0.7,0.2,162.0,24.0,20.0,6.4,3.2,1.00,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
414,65.0,Male,1.4,0.6,260.0,28.0,24.0,5.2,2.2,0.70,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


## 📊 Información de los datos, análisis y resultados

Este proyecto utiliza un dataset de pacientes con información relacionada con
la presencia de problemas hepáticos. El objetivo es desarrollar un modelo de
clasificación que permita predecir si un paciente presenta o no problemas de
hígado a partir de sus características clínicas y demográficas.

El dataset utilizado corresponde a una versión modificada por el docente para
efectos académicos y se encuentra almacenado localmente en:

`data/01_raw/Pacientes_porblemas_higado_india.csv`

### Variables del dataset

Las variables representan características demográficas y resultados de
pruebas relacionadas con la función hepática. Entre las principales variables
se encuentran:

| Variable | Descripción |
|---|---|
| `Age` | Edad del paciente en años. |
| `Gender` | Género del paciente. |
| `TB` | Bilirrubina total. |
| `DB` | Bilirrubina directa. |
| `Alkphos` | Fosfatasa alcalina. |
| `Sgpt` | Alanina aminotransferasa (ALT), enzima relacionada con la función hepática. |
| `Sgot` | Aspartato aminotransferasa (AST), enzima relacionada con la función hepática. |
| `TP` | Proteínas totales. |
| `ALB` | Albúmina. |
| `A/G Ratio` | Relación albúmina/globulina. |
| `Selector` | Variable asociada al resultado de clasificación del paciente. |

### Variable objetivo

La variable objetivo corresponde a la clasificación que indica si el paciente
presenta o no problemas hepáticos.

Esta variable será utilizada como `target` en el problema de aprendizaje
supervisado, mientras que las demás variables potencialmente relevantes serán
utilizadas como características predictoras.

La selección definitiva de las variables predictoras se realizará después de
la exploración y análisis de calidad de los datos.



### Revisión de las columnas

In [ ]:
pacientes_df.columns

Index(['Age', 'Gender', 'Total_Bilirubin', 'Direct_Bilirubin',
       'Alkaline_Phosphotase', 'Alamine_Aminotransferase',
       'Aspartate_Aminotransferase', 'Total_Protiens', 'Albumin',
       'Albumin_and_Globulin_Ratio',
       ...
       'Unnamed: 1014', 'Unnamed: 1015', 'Unnamed: 1016', 'Unnamed: 1017',
       'Unnamed: 1018', 'Unnamed: 1019', 'Unnamed: 1020', 'Unnamed: 1021',
       'Unnamed: 1022', 'Unnamed: 1023'],
      dtype='str', length=1024)

### Diagnóstico inicial de las variables

El dataset contiene un total de **1.024 columnas**. Las primeras variables corresponden a características demográficas y bioquímicas relacionadas con la función hepática:

| Variable | Descripción general |
|---|---|
| `Age` | Edad del paciente. |
| `Gender` | Género del paciente. |
| `Total_Bilirubin` | Bilirrubina total. |
| `Direct_Bilirubin` | Bilirrubina directa. |
| `Alkaline_Phosphotase` | Fosfatasa alcalina. |
| `Alamine_Aminotransferase` | Alanina aminotransferasa (ALT). |
| `Aspartate_Aminotransferase` | Aspartato aminotransferasa (AST). |
| `Total_Protiens` | Proteínas totales. |
| `Albumin` | Albúmina. |
| `Albumin_and_Globulin_Ratio` | Relación albúmina/globulina. |

Estas variables contienen información demográfica y resultados de pruebas
bioquímicas que pueden estar relacionados con la clasificación de los
pacientes.

Sin embargo, el dataset presenta **1.024 columnas**, y a partir de
`Unnamed: 1014` aparecen columnas con nombres genéricos (`Unnamed: ...`).
En esta etapa no se puede determinar únicamente por el nombre cuál es su
significado ni si contienen información útil.

Por esta razón, antes de seleccionar las variables definitivas para el modelo,
es necesario realizar un análisis de calidad que permita determinar:

- Cuáles columnas contienen información real.
- Cuáles columnas están completamente vacías.
- Cuáles contienen valores constantes.
- Qué tipo de dato tiene cada variable.
- Cuántos valores faltantes existen.
- Si existen columnas duplicadas.
- Si alguna variable contiene información relacionada directamente con el
  objetivo y puede generar *data leakage*.
- Cuál es la variable objetivo (`Dataset`) y cómo están distribuidas sus clases.

### Diagnóstico preliminar

A partir de los nombres de las variables se observa que existe un conjunto
inicial de variables clínicas claramente identificables. No obstante, la selección definitiva de variables se realizará después de completar la
exploración y limpieza del dataset.

## Planteamiento del problema

### 1. ¿Cuál es el objetivo del problema?

El objetivo es desarrollar un modelo de clasificación que permita predecir si un paciente pertenece a uno de los dos grupos definidos en la variable `Dataset`, utilizando sus características demográficas y bioquímicas.

En esta etapa inicial no se asume que las clases 1 y 2 representen específicamente presencia o ausencia de enfermedad hepática, ya que el archivo académico no documenta formalmente el significado clínico de estas clases. Este aspecto deberá confirmarse antes de interpretar los resultados del modelo desde una perspectiva clínica.

### 2. ¿Cómo se usará su solución?

La solución podría utilizarse como herramienta de apoyo para realizar una primera clasificación o priorización de pacientes que requieran una evaluación clínica adicional.

Sin embargo, por tratarse de un problema relacionado con información clínica, el modelo no debe considerarse un sistema de diagnóstico autónomo.

### 3. ¿Cuáles son las soluciones actuales?

Una alternativa existente es la evaluación clínica realizada por profesionales de la salud, quienes pueden analizar los antecedentes del paciente, sus características demográficas, resultados de pruebas de laboratorio y otros factores clínicos para determinar si se requieren evaluaciones adicionales.

### 4. ¿Cómo se debe enmarcar este problema?

El problema se puede enmarcar inicialmente como un problema de **aprendizaje supervisado de clasificación binaria**, debido a que se dispone de una variable objetivo (`Dataset`) con dos clases.

El procesamiento se realizará **fuera de línea (offline/batch)**, ya que los datos están disponibles como un archivo estático y no corresponden a un flujo de datos en tiempo real.

La preparación, validación y modelado se realizarán en etapas posteriores del proyecto.

### 5. ¿Cómo se debe medir el desempeño de la solución?

Como primera aproximación, se pueden considerar las siguientes métricas:

- Matriz de confusión.
- Precisión (*precision*).
- Sensibilidad o *recall*.
- F1-score.
- ROC-AUC.
- PR-AUC.

La selección definitiva de la métrica principal dependerá del significado de las clases y del costo asociado a los diferentes tipos de error.

### 6. ¿La medida de desempeño está alineada con el objetivo?

De manera inicial, las métricas propuestas permiten evaluar diferentes aspectos del desempeño del clasificador y son compatibles con un problema de clasificación binaria.

Sin embargo, todavía no es posible determinar cuál debe ser la métrica principal porque no se ha confirmado el significado clínico de las clases ni el costo relativo de los falsos positivos y falsos negativos.

Si posteriormente se confirma que una de las clases representa la presencia de una condición hepática, el *recall* de dicha clase podría adquirir especial importancia para reducir la cantidad de casos que el modelo no identifica.

### 7. ¿Cuál sería el desempeño mínimo necesario?

En esta etapa no es posible establecer un valor mínimo de desempeño de manera responsable.

Primero será necesario establecer un modelo *baseline*, evaluar la distribución de las clases y realizar la preparación de los datos evitando posibles problemas de *data leakage*.

Posteriormente, el desempeño mínimo deberá definirse de acuerdo con el objetivo de la solución y el costo de los diferentes tipos de error.

### 8. ¿Cuáles son los problemas parecidos? ¿Se pueden reutilizar experiencias o herramientas?

El problema es comparable con otros problemas de clasificación de riesgo en salud que utilizan información clínica y biomarcadores, como la identificación de riesgo renal, metabólico o cardiovascular.

Se pueden reutilizar herramientas y prácticas comunes de estos problemas, como:

- Preparación e imputación de datos.
- Validación cruzada.
- Regresión logística.
- Árboles de decisión.
- Modelos de *boosting*.
- Métricas para problemas de clasificación.

La elección definitiva de las técnicas se realizará después de explorar las características del dataset.

### 9. ¿Hay experiencia del problema disponible?

El dataset no proporciona información sobre una solución previa desarrollada específicamente para este ejercicio.

Sin embargo, existen metodologías y herramientas aplicables a problemas similares de clasificación con datos clínicos tabulares que pueden ser utilizadas como referencia durante las etapas de preparación, validación y modelado.

### 10. ¿Cómo se puede resolver el problema manualmente?

De manera manual, un profesional podría analizar las características demográficas del paciente y los resultados de las pruebas bioquímicas disponibles, compararlos con los rangos de referencia correspondientes e integrar esta información con los antecedentes y demás información clínica disponible.

A partir de esta evaluación, el profesional podría determinar si se requiere una evaluación adicional.

El modelo buscaría automatizar la combinación de las variables disponibles para realizar la clasificación definida por el dataset, pero no sustituiría la evaluación clínica profesional.

### 11. Supuestos actuales

En esta etapa se establecen los siguientes supuestos:

- `Dataset` corresponde a la variable objetivo que se desea predecir.
- La variable objetivo contiene dos clases.
- Las dos clases representan grupos diferentes de pacientes.
- Las variables clínicas y demográficas contienen información útil para realizar la clasificación.
- Los registros corresponden a observaciones independientes de pacientes.
- El dataset se encuentra disponible como un archivo estático.
- El significado clínico exacto de las clases debe confirmarse antes de realizar interpretaciones médicas.
- El costo de los falsos positivos y falsos negativos aún no está definido.
- La selección definitiva de variables y el tratamiento de posibles problemas de *data leakage* se realizarán durante las etapas posteriores de análisis.

### 12. ¿Cuál es la fuente de los datos?

La fuente de referencia del dataset base es el **UCI Machine Learning Repository**, específicamente el *Indian Liver Patient Dataset (ILPD)*.

Para efectos académicos, el docente proporcionó una versión modificada del dataset original. Esta versión modificada es la que se utiliza exclusivamente en este proyecto.

El archivo utilizado se encuentra en:

`data/01_raw/Pacientes_porblemas_higado_india.csv`

El dataset original de UCI no se descarga ni se utiliza directamente en el proyecto.

### 13. ¿Cómo se actualizan los datos?

El dataset académico utilizado en este proyecto es un archivo local estático. No se dispone de información sobre un mecanismo automático de actualización ni sobre un proceso periódico de incorporación de nuevos registros.

Cualquier modificación del dataset dependería de una nueva versión o entrega del archivo académico por parte del responsable del ejercicio.

### 14. ¿Cada cuánto tiempo se actualizan los datos?

No se encuentra documentada una frecuencia de actualización para el dataset académico proporcionado.
Por esta razón, para este proyecto se considera inicialmente como un **dataset estático** empleado para este ejercicio.